### PR TITLE
Hotel Katherine Fitzpatrick

### PR DESCRIPTION
**Pull request by KatherineJF**
# Hotel
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe a design decision you had to make when working on this project. What options were you considering? What helped you make your final decision? |  How to build the classes. Decided that room and reservation should describe only methods and variables specific to them and that the room_tracker class would act as the hotel "reception desk" by calculating and finding availability.

Describe a concept that you gained more clarity on as you worked on this assignment. | The importance of testing and how it should inform design. 

Describe a _nominal test_ that you wrote for this assignment. |  One of the nominal tests that I wrote would be checking to see if I can create a new instance of a class like Room.

Describe an _edge case test_ that you wrote for this assignment. | An edge case I would write would involve how to calculate cost if somebody checks-in and checks-out on the same day.

How do you feel you did in writing pseudocode first, then writing the tests and then the code? | I thought this was a very helpful process. Thinking about the logic first helped me separate figuring out syntax and working on the logic.

### CODE REVIEW

File: spec/reservation_spec.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+       reservation_test = Reservation.new(start_date, end_date, room)
+
+
+         expect(reservation_test.calculate_total_cost).must_equal 400
+end
+end
+
+
+end
+end
```
Comment (CheezItMan) [inline spec/reservation_spec.rb:37]: You are missing an `end` here. Also indentation issues.

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -1,8 +1,11 @@
 require 'minitest'
-require 'minitest/autorun'
 require 'minitest/reporters'
+require 'minitest/autorun'
 # Add simplecov
```
Comment (CheezItMan) [inline spec/spec_helper.rb:4]: You haven't added `simplecov`

File: lib/room.rb [hunk 3]
```diff
@@ -0,0 +1,78 @@
+#Room Class
```
Comment (CheezItMan) [inline lib/room.rb:1]: This class isn't doing much other than providing an `id`.  So... maybe just use an array of numbers for the rooms.

File: lib/reservation.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+  def calculate_total_cost()
+    total_cost = 0
+    number_of_days = ((end_date - start_date).to_i) - 1
+    if number_of_days < 1
+      number_of_days = 1
+    end
+    total_cost = number_of_days * 200
+    return total_cost
+    end
```
Comment (CheezItMan) [inline lib/reservation.rb:29]: indentation

File: lib/reservation.rb [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+  attr_reader :start_date, :end_date, :room, :total_cost
+
+  def initialize(start_date, end_date, room)
+    @start_date = start_date
+    @end_date = end_date
+    @room = room
+    @total_cost = calculate_total_cost
+  end
+
```
Comment (CheezItMan) [inline lib/reservation.rb:20]: I would also suggest you add a method to tell if two reservations overlap, and another to tell if a reservation occurs on a specific date.

File: lib/room_tracker.rb [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+    end
+    return rooms
+  end
+
+  def list_all_rooms
+    return @rooms
+  end
+
+  def make_reservation(start_date, end_date)
+    room = @rooms[rand(@rooms.length)] #getavilable room method
```
Comment (CheezItMan) [inline lib/room_tracker.rb]: You should limit this to rooms that are available in the given date range.

File: lib/room_tracker.rb [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+    @bookings << Reservation.new(start_date, end_date, room)
+  end
+
+  def find_reservations_by_date(date)
+    bookings = []
+    @bookings.each do |booking|
+      range = booking.start_date..booking.end_date
```
Comment (CheezItMan) [inline lib/room_tracker.rb:77]: This whole check to see if a Reservation occurs on that date would be wonderful as a helper method in `Reservation`.

File: spec/room_tracker_spec.rb [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+  end
+end
+
+  describe "I can reserve a room for a given date range" do
+    describe "#make_reservation" do
+      let (:start_date) { Date.parse('2012-02-01') }
+      let (:end_date) { Date.parse('2012-02-03') }
+      let (:room) { [14] }
+      let (:bookings) {[]}
+             it "assigns available room" do
```
Comment (CheezItMan) [inline spec/room_tracker_spec.rb:53]: indentation

File: spec/room_tracker_spec.rb [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+               hotel = Room_Tracker.new()
+               hotel.make_reservation(start_date,end_date)
+               new_reservation = hotel.bookings.last
+               #new_booking = make_reservation(start_date,end_date, room)# new_bookings = Reservation.new(start_date,end_date, 12)
+               expect (new_reservation.room).must_equal 14
```
Comment (CheezItMan) [inline spec/room_tracker_spec.rb:58]: How do you know it's 14.  The reservation picks a random room #.

**Comment by CheezItMan** [comment]
# Hotel
## What We're Looking For

Feature | Feedback
---     | ---
**Baseline** |
Used git regularly	| Reaonable number of commits, but don't use waves in your commit messages.  Instead describe the functionality you added.  
Answer comprehension questions	| Check, glad you found it a helpful process.
**Design** |
Each class is responsible for a single piece of the program | Check
Classes are loosely coupled | Check
**Wave 1** |
List rooms | Check
Reserve a room for a given date range | Check, but it doesn't check to see if that room is available.
List reservations for a given date | Check
Calculate reservation price | Check
Invalid date range produces an error | Check
Test coverage | `SimpleCov` not added
**Wave 2** |
View available rooms for a given date range | **MISSING**
Reserving a room that is not available produces an error | **MISSING**
**Wave 3** |  **MISSING**
Create a block of rooms | 
Check if a block has rooms | 
Reserve a room from a block | 
**Fundamentals** | 
Names variables, classes and modules appropriately | 
Understanding of variable scope - local vs instance | 
Can create complex logical structures utilizing variables | 
Appropriately uses methods to break down tasks into smaller simpler tasks | 
Understands the differences between class and instance methods | 
Appropriately uses iterators and enumerables | 
Appropriately writes and utilizes classes | 
Appropriately utilizes modules as a mechanism to encapsulate functionality | 
**Wrap Up** |
There is a refactors.txt file | **MISSING**
The file provides a roadmap to future changes | **MISSING**
**Additional Feedback** | You got through wave 1, but weren't able to figure out how to handle reservations and select rooms that are available in that time period.  The basic design (classes, and their relationships) is pretty good however.  I also left some notes in your code and let me know if you have questions.  

One more minor note.  Please don't brake the markdown table in the comprehension questions.
